### DISCUSSION TITLE
Get a list of all repositories where I'm a collaborator/invited

### DISCUSSION THREAD

**Discussion by aram535** [community/community#24514] [Apps, API and Webhooks]
<p>Maybe someone can point me to where I’m going wrong here …</p>
<p>API v3 documentation <a href="https://developer.github.com/v3/repos/" rel="nofollow noopener">https://developer.github.com/v3/repos/</a></p>
<p>I have a user with access to three (3) repositories:</p>
<p>R1 - NEW empty repo (testRepository)<br>
R2 - Fork of a public repo (github-api)<br>
R3 - private repository of another user which I have been invited to (and accepted).</p>
<p>However, the API only return R1 and R2?</p>
<pre><code>curl -X GET https://api.github.com/users/&lt;myuser&gt;/repos\?type=all
</code></pre>
<p>How do I get a list of all repositories where I’m not the owner but just a collaborator/invited?</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/aram535">@aram535</a>,</p>
<p>Thank you for being here! I wonder if you are running into a pagination issue, could you try:</p>
<pre><code>curl -X GET https://api.github.com/users/&lt;myuser&gt;/repos?per_page=100
</code></pre>
<p>Results are returned in pages of 30 results by default, so you can follow the Link header to get the next page of results, and you can also add the per_page parameter to get more results.</p>

**Comment by aram535** [comment]
<p>Hi Andrea, </p>
<p>It’s a rather small request at this point, only 3 should return.  “type=all” should include any repositories that the user has been invited into, however only the 2 “owned” projects are returned.</p>
<p>Here is the odd part:</p>
<pre><code>$ curl -s "https://api.github.com/user/repos?type=all" | jq '.'
3
</code></pre>
<p>but if you query the user itself:</p>
<p>Only 2 repos are returned.</p>
<p>I have verified it manually… and here it is programmatically:</p>
<pre><code>$ curl -s "https://api.github.com/users/${USER}/repos?type=all&amp;per_page=100" -H "Authorization: token ${AUTH}" | jq '. | length'
2
</code></pre>
<p>Just as a test using V4 of the API I do get 3 repositories.</p>
<pre><code>{
  viewer {
    name
    repositories(first: 5) {
      edges {
        node {
          id
          name
        }
      }
    }
  }
}
</code></pre>
<p>I get 3 repositories listed. </p>

**Reply by Jon104** [reply]
So using the V4 with graphql, you successfully listed another user repository you were invited to contribute?
I'm trying to do the same thing in v3, but it looks like it's a deadend... 

Sad i have no clue how v4 works :D! I'll take a look !

**Reply by Jon104** [reply]
I just did some tests, and from the GraphQL Explorer, where you sign up and tests queries. I do get the other user repository.
So then i tried it in my javascript server, and it still doesn't return the repository i need... :( 
Not sure what type of authentification is used in that GraphQL explorer...

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/aram535">@aram535</a>,</p>
<p>Are all the memberships public? If you don’t authenticate, that endpoint returns only organizations for which you publicized your membership, as explained in the documentation:</p>
<p><a href="https://developer.github.com/v3/orgs/#list-user-organizations" rel="nofollow noopener">https://developer.github.com/v3/orgs/#list-user-organizations</a></p>
<p>Thought that doesn’t really explain how v4 is returning them all, let me look into this some more. I’ll reach out to that team and post any updates here. Thanks again for bringing this up to our attention.</p>
<p>Best,</p>
<p>Andrea</p>

**Comment by aram535** [comment]
<p>Hi Andrea, </p>
<p>No, they’re not all public.  Yes, I am authenticating using a token (same token).  I don’t believe without authentication the 2nd endpoint would return anything at all correct?</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/aram535">@aram535</a>,</p>
<p>You are correct, the authenticated user has explicit permission to access repositories they own, repositories where they are a collaborator, and repositories that they can access through an organization membership. </p>
<p>It would seem the simplest way in v3 is to use /user/repos: <a href="https://api.github.com/user/repos" rel="nofollow noopener">https://api.github.com/user/repos</a>.</p>
<p>I hope this helps!</p>
<p>Best,</p>
<p>Andrea</p>

**Reply by thewh1teagle** [reply]
> You are correct, the authenticated user has explicit permission to access repositories they own, repositories where they are a collaborator, and repositories that they can access through an organization membership. 
> 
> It would seem the simplest way in v3 is to use /user/repos: https://api.github.com/user/repos.

I tried to use this API endpoint with the following command:

```console
gh api -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" "/users/user/repos?type=all"
```

But it doesn't include repositories shared with me. only the repositories I created.

**Comment by jonathanhle** [comment]
<p>Is a Github Admin (within Github Enterprise), able to pull up all the memberships (public/private) for specific members?</p>
<p>I’ve been banging my head against the V3 apis and can’t see to get all the repos returned, when I know a specific user (not me) has write access to it.</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/jonathanhle">@jonathanhle</a>, welcome to the community! Thanks for your question. Users can <a href="https://help.github.com/enterprise/user/articles/publicizing-or-hiding-organization-membership/">publicize or hide organization membership</a>. Hidden organization memberships are not visible in the GitHub Enterprise web interface unless your user, too, is a member of the organization, and they never appear in responses from the Organizations API’s <a href="https://developer.github.com/enterprise/v3/orgs/#list-user-organizations"> <em>List user organizations</em>  endpoint</a>.</p>
<p>In order to see someone’s hidden organization membership as a Site Administrator via the API, you’ll need to make a request to the Members API’s <a href="https://developer.github.com/enterprise/v3/orgs/members/#list-your-organization-memberships"> <em>List your organization memberships</em>  endpoint</a> and authenticate that request with an <a href="https://developer.github.com/enterprise/v3/users/administration/#create-an-impersonation-oauth-token">impersonation OAuth token</a>. This will show all organization memberships, both public and hidden. When you’re done, we advise <a href="https://developer.github.com/enterprise/v3/users/administration/#delete-an-impersonation-oauth-token">deleting the impersonation OAuth token</a>.</p>

[a comment by an automated account is not shown]
